File: app/src/main/java/com/codemybrainsout/imageviewer/utility/ImageUtils.java
package com.codemybrainsout.imageviewer.utility;

import android.content.Context;
import android.graphics.Bitmap;
import android.support.v8.renderscript.Allocation;
import android.support.v8.renderscript.Element;
import android.support.v8.renderscript.RenderScript;
import android.support.v8.renderscript.ScriptIntrinsicBlur;

/**
 * Created by ahulr on 10-06-2017.
 */

public class ImageUtils {

    public static Bitmap blur(Context context, Bitmap image, int mRadius) {
        if (null == image) return null;

        Bitmap outputBitmap = Bitmap.createBitmap(image);
        final RenderScript renderScript = RenderScript.create(context);
        Allocation tmpIn = Allocation.createFromBitmap(renderScript, image);
        Allocation tmpOut = Allocation.createFromBitmap(renderScript, outputBitmap);

        //Intrinsic Gausian blur filter
        ScriptIntrinsicBlur theIntrinsic = ScriptIntrinsicBlur.create(renderScript, Element.U8_4(renderScript));
        theIntrinsic.setRadius(mRadius);
        theIntrinsic.setInput(tmpIn);
        theIntrinsic.forEach(tmpOut);
        tmpOut.copyTo(outputBitmap);
        return outputBitmap;
    }

}

File: app/src/main/java/com/codemybrainsout/imageviewer/utility/ImageUtils.kt
package com.codemybrainsout.imageviewer.utility

import android.content.Context
import android.graphics.Bitmap
import android.icu.util.RangeValueIterator
import android.renderscript.Allocation
import android.renderscript.Element.U8_4
import android.renderscript.RenderScript
import android.renderscript.ScriptIntrinsicBlur

/**
 * Created by ahulr on 10-06-2017.
 */
object ImageUtils {

    fun blur(context: Context?, image: Bitmap?): Bitmap? {
        val bitmap = image?.copy(image.config, true)
        val rs = RenderScript.create(context)
        val input = Allocation.createFromBitmap(
            rs,
            bitmap,
            Allocation.MipmapControl.MIPMAP_FULL,
            Allocation.USAGE_SCRIPT
        )
        val output = Allocation.createTyped(rs, input.type)
        val script = ScriptIntrinsicBlur.create(rs, U8_4(rs))
        script.setRadius(10f)
        script.setInput(input)
        script.forEach(output)
        output.copyTo(bitmap)
        return bitmap
    }
}